### PR TITLE
[AD-302] Use `resolveWalletRefThenRead` and similar functions where appropriate.

### PR DESCRIPTION
**Pull request by Haskell-mouse**
**YT issue:** https://issues.serokell.io/issue/AD-302

**Description:**

1. Added function `resolveAccountRefThenRead` and
2. Now `resolveWalletRefThenRead` is mostly used instead of `resolveWalletRef`,
    because we mostly used it to read somthing from wallet.

### CODE REVIEW

File: ariadne/cardano/src/Ariadne/Wallet/Backend/KeyStorage.hs [hunk 1]
```diff
@@ -467,34 +461,25 @@ select
   -> IO ()
 select acidDb WalletFace{..} walletSelRef mWalletRef uiPath = do
   walletDb <- query acidDb Snapshot
-  let wallets = walletDb ^. dbHdWallets
   mbSelection <- case mWalletRef of
     Nothing -> return Nothing
     Just walletRef -> do
       -- Throw an exception if walletRef is invalid
-      rootId <- resolveWalletRef walletSelRef walletRef walletDb
+      (rootId,accList) <- getCompose $ toList <$> (Compose $
```
Comment (gromakovsky) [inline ariadne/cardano/src/Ariadne/Wallet/Backend/KeyStorage.hs]: `second toList <$> resolveWalletRefThenRead walletSelRef walletRef walletDb readAccountsByRootId` please.

**Comment by gromakovsky** [comment]
Also there are trailing whitespaces on some lines. It's recommended to setup your editor to remove them automatically.

**Comment by gromakovsky** [comment]
> I'll also check whether there are other places where resolveWalletRefThenRead or resolveAccountRefThenRead can be used.

Apparently there are no other such places.

File: ariadne/cardano/src/Ariadne/Wallet/Backend/KeyStorage.hs [hunk 2]
```diff
@@ -208,7 +207,8 @@ resolveAccountRef walletSelRef accountRef walletDb = case accountRef of
       checkParentRoot accId
       return accId
     AccountRefByUIindex accIdx walRef -> do
-      accounts <- getAccList walRef
+      accounts <- snd <$> (getCompose $ toList <$> (Compose $
```
[the remark this answers is not in the document]
Comment (Haskell-mouse) [reply ariadne/cardano/src/Ariadne/Wallet/Backend/KeyStorage.hs]: Ok, i agree, just used to trying to use more general things. In this case we don't use it's functor propertis anyumore, and we don't need to use compose.  (in other case using compose can be more refactorable). 
Ok, will change.

**Comment by gromakovsky** [comment]
And now `Compose` is used as it was initially :man_shrugging:

**Pull request merged**